### ISSUE TITLE
CAPO request to create a already existing VM 

### ISSUE THREAD

**Issue by hugoprudente** [kubernetes-sigs/cluster-api-provider-openstack#2181]
/kind bug

**What steps did you take and what happened:**

Cluster was running happy there was not reboot on the June 28 VM but somehow I got a trigger and it created another VM. 

```
dub-md-2-78fkp centos-9-kube-v1.28.8-g8912.2 Sept. 10, 2024, 10:25 p.m.
dub-md-2-78fkp centos-9-kube-v1.28.8-g8912.2 June 28, 2024, 11:22 a.m.
```

**What did you expect to happen:**

No duplicate VMs if the VM is not deleted from Openstack side. 

**Anything else you would like to add:**

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): v0.9.0
- Cluster-API version: v1.6.3
- OpenStack version: Yoga
- Minikube/KIND version: 
- Kubernetes version (use `kubectl version`):1.28.8
- OS (e.g. from `/etc/os-release`): centos-9 stream

**Comment by EmilienM** [comment]
please provide CAPO logs (sanitized with no company data) so we can see what trigrered that.

**Comment by EmilienM** [comment]
I think this was solved in v0.11 with OpenStackServer therefore we wouldn't spend much time fixing that in 0.9 unless it's an issue one would hit frequently.

**Comment by hugoprudente** [comment]
Thanks! It didn't happened again since I opened the issue so I could get fresh logs. 

I"m on the path to update to 0.11 thanks.
